posthog/models/notification_setting.py

```diff
@@ -0,0 +1,46 @@
+from enum import Enum
+from typing import Optional
+
+from posthog.models import User
+
+
+class NotificationSetting(Enum):
+    WEEKLY_PROJECT_DIGEST = "weekly_project_digest"
+    PLUGIN_DISABLED = "plugin_disabled"
+    # Add more notification types as needed
+
+
+def should_send_notification(
```

Comment:
This isn't a mode so feels odd to be in a new model file. This might work better as a user model method.